amads/time/tempo.py

```diff
@@ -1 +1,213 @@
+
+    where:
+
+    .. math::
+        \sigma = \sqrt{\frac{\sum\limits_{i=1}^{N} (y_i - (\hat{S} x_i + \hat{b}))^2}{N - 2}},
+
+    and:
+
+    - :math:`x_i` is the time of beat :math:`i`.
+    - :math:`y_i` is the tempo value in quarter-note beats-per-minute.
+    - :math:`\hat{S}` is the estimated slope of the regression.
+    - :math:`\hat{b}` is the intercept.
+    - :math:`N` is the number of data points.
+
+    Args:
+        beats (Iterable[float]):
+            An iterable of beat timestamps in seconds, such as quarter-note onsets.
+
+    Returns:
+        float: The computed tempo drift value.
+
+    """
+
+    return float(fit_tempo_linear_regression(beats).slope_stderr)
+
+
+# flake8: noqa: W605
+def tempo_fluctuation(beats: Iterable[float]) -> float:
+    r"""
+    Calculates the percentage fluctuation around the overall tempo of a sequence of beats.
+
+    Tempo fluctuation is measured as the standard deviation of the instantaneous tempo,
+    normalized by the mean tempo. Higher values indicate greater variability in tempo.
+
+    The equation is:
+
+    .. math::
+        \text{F} = \dfrac{\sqrt{\frac{1}{N-1} \sum\limits_{i=1}^N (y_i - \bar{y})^2}}{\bar{y}},
+
+    where :math:`y_i` is the tempo value in quarter-note beats-per-minute at beat :math:`i`.
+
+    Args:
+        beats (Iterable[float]):
+            An iterable of beat timestamps in seconds, such as quarter-note onsets.
+
+    Returns:
+        float: The computed tempo fluctuation value.
+
+    """
+
+    # Compute instantaneous tempo values: this will also validate the input
+    tempos = beats_to_tempo(beats)
+    # Compute tempo fluctuation
+    # No need for np.nanstd etc. here, we won't have NaN values
+    return float(np.std(tempos) / np.mean(tempos))
+
+
+def _validate_beats(beats: Union[Iterable[float], np.ndarray]) -> None:
+    """Checks an iterable of beat timestamps and raises errors as required"""
+
+    if not isinstance(beats, np.ndarray):
+        beats = np.array(beats, dtype=float)  # expecting a float dtype
+    if np.isnan(np.sum(beats)):
+        raise ValueError(
+            "Cannot calculate instantaneous tempo measurements from array of beats with missing values"
+        )
+    if len(beats) < 2:
+        raise ValueError("Must pass at least two beat timestamps")
+    if len(beats.shape) != 1:
+        raise ValueError("Beat timestamps must be 1-dimensional")
+    if not np.array_equal(np.sort(beats), beats):
+        raise ValueError("Beat timestamps must increase linearly")
+    if len(np.unique(beats)) != len(beats):
+        raise ValueError("Beat timestamps must not contain duplicates")
+
+
+def beats_to_tempo(beats: np.ndarray) -> np.ndarray:
+    """Converts beat timestamps to instantaneous tempo measurements."""
+
+    # Raise error on invalid inputs
+    _validate_beats(beats)
+    return np.array(60 / np.diff(beats))
+
+
+def linregress(x: np.ndarray, y: np.ndarray) -> LinearRegressionWrapper:
```

Comment:
I appreciate the enthusiasm but given that we already have scipy as a dependency I think it would be safest to use `scipy.stats.lingress` here!